package/gluon-offline-ssid/luasrc/usr/bin/gluon-offline-ssid

```diff
@@ -0,0 +1,241 @@
+    os.exit(0)
+  end
+else
+  local MSG = ""
+  CHECK=tonumber(shell_exec('batctl gwl -H | grep -v "gateways in range" -c'))
+end
+
+local UP = UT / 60
+local M = UP % MINUTES
+
+local HUP_NEEDED = 0
+
+if CHECK > 0 or DISABLED == '1' then
+  print("node is online")
+  -- check status for all physical devices
+  -- TODO: real loop over phy
+  for SSID in SSIDs do
+    CURRENT_SSID=shell_exec('grep "^ssid=' .. OFFLINE_SSID .. '" ' .. HOSTAPD .. ' | cut -d"=" -f2)"')
+    if CURRENT_SSID == ONLINE_SSID then
+      print("SSID $CURRENT_SSID is correct, nothing to do")
+      break
+    end
+    if CURRENT_SSID == OFFLINE_SSID then
+      -- set online
+      shell_exec('logger -s -t "gluon-offline-ssid" -p 5 ' .. MSG .. '"SSID is ' .. CURRENT_SSID .. ', change to ' .. ONLINE_SSID .. '"')
+      shell_exec('sed -i "s~^ssid=' .. CURRENT_SSID .. '~ssid=' .. ONLINE_SSID .. '~" ' .. HOSTAPD .. '')
+      -- HUP here would be to early for dualband devices
+      HUP_NEEDED=1
+    else
+      shell_exec('logger -s -t "gluon-offline-ssid" -p 5 "could not set to online state: did neither find SSID ' .. ONLINE_SSID .. ' nor ' .. OFFLINE_SSID .. '. Please reboot"')
+    end
+  end
+elseif CHECK == 0 then
+  print("node is considered offline")
+  if UP < FIRST or M == 0 then
+    -- set SSID offline, only if uptime is less than FIRST or exactly a multiplicative of switch_timeframe
+    if UP < FIRST then 
+      T=FIRST
+    else
+      T=MINUTES
+    end
+    -- print("Minute M, check if OFF_COUNT is more than half of " .. T )
+    if OFF_COUNT >= T / 2 then
+      -- node was offline more times than half of switch_timeframe (or than FIRST)
+      -- TODO: real loop over phy
+      for HOSTAPD in phys do
+        ONLINE_SSID=shell_exec(ONLINE_SSIDs .. ' | awk -F \'~\' -v l=' .. (LOOP*2) .. ' \'{print $l}')
+        CURRENT_SSID=shell_exec('grep "^ssid=' .. OFFLINE_SSID .. '" ' .. HOSTAPD .. ' | cut -d"=" -f2)"')
+        if CURRENT_SSID == OFFLINE_SSID then
+          print("SSID ' .. CURRENT_SSID .. ' is correct, nothing to do")
+          break
+        end
+        CURRENT_SSID=shell_exec('grep "^ssid=' .. ONLINE_SSID .. '" ' .. HOSTAPD .. ' | cut -d"=" -f2)"')
+        if CURRENT_SSID == ONLINE_SSID then
+          -- set offline
+          shell_exec('logger -s -t "gluon-offline-ssid" -p 5 ' .. MSG .. '"' .. OFF_COUNT .. ' times offline, SSID is ' .. CURRENT_SSID .. ', change to ' .. OFFLINE_SSID .. '"')
+          shell_exec('sed -i "s~^ssid=' .. ONLINE_SSID .. '~ssid=' .. OFFLINE_SSID .. '~" ' .. HOSTAPD .. '')
+          HUP_NEEDED=1
+        else
+          shell_exec('logger -s -t "gluon-offline-ssid" -p 5 "could not set to offline state: did neither endnd SSID ' .. ONLINE_SSID .. ' nor ' .. OFFLINE_SSID .. '. Please reboot"')
+        end
+      end
+    end
+    -- else print("Minute ' .. M .. ', just count ' .. OFF_COUNT .. '")
+  end
+  
+  local f = io.open(TMP, 'w+')
+  f:write(OFF_COUNT + 1)
```

Comment:
io.popen(tmp, 'w'):write(off_count + 1)
The capital variable names look pretty in shell, but not in Lua.

Furthermore your config file is /etc/config/gluon-offline-ssid. Thus you have to use uci get gluon-offline-ssid instead of uci get offline-ssid.

Reply:
only `w`? without `+`?

Reply:
You don't close the file, but assign a file pointer, you don't use assert, you open the file in read-write mode instead of write-only mode while never reading the file.

Reply:
yes, seems to work with only `w`